content/public/android/java/src/org/chromium/content/browser/MediaResourceGetter.java

```diff
@@ -176,33 +277,18 @@ boolean isValidHlsUrl(String url) {
     }
 
     @VisibleForTesting
-    List<String> getHlsMetadataRequest(String url) {
+    List<String> getHlsMetadataRequest(final Context context, String url) {
         if (isValidHlsUrl(url)) {
-            BufferedReader reader = null;
-            List<String> lines = new ArrayList<>();
-            try {
-                HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
-                connection.setRequestMethod("GET");
-                connection.connect();
-                reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
-                String line = null;
-                while ((line = reader.readLine()) != null) {
-                    line = line.trim();
-                    if (!line.startsWith("##") && !line.equals("#EXTM3U")) {
-                        lines.add(line);
-                    }
-                }
-                return lines;
-            } catch (IOException e) {
-                Log.e(TAG, "Cannot get hls metadata from: %s", url);
-            } finally {
-                if (reader != null) {
-                    try {
-                        reader.close();
-                    } catch (IOException e) {
-                    }
-                }
-            }
+            UrlRequestContextConfig myConfig = new UrlRequestContextConfig();
```

Comment:
Concerned that this entire block isn't at least wrapped in a base try/catch block.  I'd think many parts of it could potentially fail.

Reply:
I double checked the files under Cronet Src @ https://github.com/nv-chromium/chromium-crosswalk/tree/nv-crosswalk-17/components/cronet and none of the methods used in this code block are throwing any exceptions to be caught. Never even saw a compilation error related to exceptions.

Do you suggest I just put a try catch(Exception) with the generic exception?

Reply:
I'd wrap the entirety of 282-291 in a try/catch block.  Didn't mean just that one line, something could fail in any part of that block.